### PR TITLE
Divyanshu | Stage | "Quantity recorded can't be more than order" error coming in updating PB for stock and updating qty when no PO linked

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cwnehh0

### CODE REVIEW

File: apps/web-giddh/src/app/vouchers/create/create.component.ts [hunk 1]
```diff
@@ -2331,6 +2331,28 @@ export class VoucherCreateComponent implements OnInit, OnDestroy, AfterViewInit
         });
     }
 
+    /**
+     * Calculate max quantity for PO linking in PB 
+     * 
+     * @param entryData 
+     * @returns 
+     */
+    private getStockMaxQuantity(entryData): number | undefined {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/create/create.component.ts]: Add type for entryData

File: apps/web-giddh/src/app/vouchers/create/create.component.ts [hunk 2]
```diff
@@ -2331,6 +2331,28 @@ export class VoucherCreateComponent implements OnInit, OnDestroy, AfterViewInit
...
+        let maxQuantity = undefined;
+        if (this.invoiceType.isPurchaseInvoice && entryData.purchaseOrderLinkSummaries?.length > 0) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/create/create.component.ts]:
```suggestion
        if (this.invoiceType.isPurchaseInvoice && entryData?.purchaseOrderLinkSummaries?.length > 0) {
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vouchers/create/create.component.ts (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vouchers/create/create.component.ts]: Done

File: apps/web-giddh/src/app/vouchers/create/create.component.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vouchers/create/create.component.ts]: No need here because entryData will always  present

**Review by dilpreetsinghofficial — DISMISSED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**